definitions/features/foreman_proxy.rb

```diff
@@ -209,11 +209,23 @@ def lookup_dhcpd_config_file
     dhcpd_config_file
   end
 
-  def lookup_using_dhcp_yml
-    dhcp_yml_path = lookup_into(FOREMAN_PROXY_DHCP_YML_PATHS)
-    raise "Couldn't find dhcp.yml file under foreman-proxy" unless dhcp_yml_path
+  def dhcp_yml_path
+    dhcp_path = lookup_into(FOREMAN_PROXY_DHCP_YML_PATHS)
+    raise "Couldn't find dhcp.yml file under foreman-proxy" unless dhcp_path
+
+    dhcp_path
+  end
+
+  def parse_dhcp_yml
+    yaml_load(dhcp_yml_path)
+  end
+
+  def dhcp_provider
+    parse_dhcp_yml[:use_provider]
+  end
 
-    configs_from_dhcp_yml = yaml_load(dhcp_yml_path)
+  def lookup_using_dhcp_yml
+    configs_from_dhcp_yml = parse_dhcp_yml
```

Comment:
With above change, dhcp_yml configs will be accessible directly using `configs_from_dhcp_yml`